tests/unit_tests/dft/include/compute_inplace.hpp

```diff
@@ -152,6 +151,12 @@ int DFT_Test<precision, domain>::test_in_place_buffer() {
                                  complex_strides.data());
             descriptor.set_value(oneapi::mkl::dft::config_param::OUTPUT_STRIDES,
                                  real_strides.data());
+            // need to do this because for mklcpu because classical API does have the mechanism to utilize the 
+            // XWD_DISATANCES to not reset the distance parameters
+            if(dev->is_cpu()) {
+                descriptor.set_value(oneapi::mkl::dft::config_param::FWD_DISTANCE, backward_elements);
+                descriptor.set_value(oneapi::mkl::dft::config_param::BWD_DISTANCE, container_size_per_transform);
+            }
```

Comment:
This isn't very good, how will the user know to do this?
The way we've got around it in the cuFFT backend is to have two "plans" for each descriptor, one for going forward and one for going backward.

Reply:
This makes the library unusable for batched transforms. A compromise could be to disable batched transforms until you fix this.

Reply:
fixed by using a bi-directional handle container : 40cabd8bfdac4f97570f911f503cb752cfe349e0, 113cb04c8b4ee7491ca33b94795c6ffb044c2d2f